contracts/test/PoolSwapTest.sol

```diff
@@ -35,21 +36,29 @@ contract PoolSwapTest is ILockCallback {
         payable
         returns (BalanceDelta delta)
     {
-        delta =
-            abi.decode(manager.lock(abi.encode(CallbackData(msg.sender, testSettings, key, params))), (BalanceDelta));
+        return swap(key, params, testSettings, new bytes(0));
```

Comment:
looks like gas went up by >1k for all swaps...wonder if consolidating the swap function here makes more sense than having 2 external calls (or just keeping them separate)

Reply:
I prefer removing the old function signature and moving everything over to the new function signature w/ `bytes memory hookData` (instead of duplicating the logic). I wanted to keep the original for backwards compatibility with v4-periphery and community repos, but I suppose theres been breaking changes already so might as well keep it up 😝 

Will try it for both swap + modifyPosition